ubuntu-github-actions-android/Dockerfile-30.0.3

```diff
@@ -0,0 +1,71 @@
+# Version: 20231114
+# Image name: quay.io/opencv-ci/opencv-androidsdk-30
+# For 4.x branch
+
+FROM ubuntu:20.04
+
+ARG GRADLE_VERSION=7.5.1
+ARG ANDROID_BUILD_TOOLS_VERSION=30.0.3
+ARG ANDROID_PLATFORM_VERSION=android-26
+ARG ANDROID_TOOLS_ARCHIVE=commandlinetools-linux-6858069_latest.zip
+ARG NDK_VERSION=18.1.5063045
+ARG CMAKE_VERSION=3.22.1
+
+USER root
+
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN \
+  apt-get update && \
+  apt-get install -y --no-install-recommends \
+    software-properties-common \
+    build-essential ccache ninja-build \
+    python3 python3-pip curl zip unzip file \
+    openjdk-8-jdk openjdk-17-jre libncurses5 tree \
+    && \
+  add-apt-repository ppa:git-core/ppa && \
+  apt-get install -y --no-install-recommends git && \
+  rm -rf /var/lib/apt/lists/*
+
+RUN mkdir -p /opt/gradle && \
+    curl -L -o /opt/gradle/gradle-$GRADLE_VERSION-bin.zip https://services.gradle.org/distributions/gradle-$GRADLE_VERSION-bin.zip && \
+    unzip -d /opt/gradle /opt/gradle/gradle-$GRADLE_VERSION-bin.zip
+
+ENV ANDROID_HOME="/opt/android-sdk"
+
+# find+chmod fixes permission issue for old tools package
+RUN cd ~ && mkdir $ANDROID_HOME .android tmp \
+    && cd tmp \
+    && curl -o sdk.zip "https://dl.google.com/android/repository/$ANDROID_TOOLS_ARCHIVE" \
+    && unzip sdk.zip \
+    && rm sdk.zip \
+    && find $ANDROID_HOME -executable -exec chmod a+rx {} \; \
+    && mkdir -p $ANDROID_HOME/cmdline-tools/latest \
+    && mv cmdline-tools/* $ANDROID_HOME/cmdline-tools/latest/ \
+    && tree $ANDROID_HOME \
+    && echo "y\n" > /tmp/yes \
```

Comment:
I wrote this script for installing SDK some time ago: https://github.com/mshabunin/opencv-android-build/blob/master/scripts/get_sdk.sh
* might be necessary to run `sdkmanager --update` first
* you can run `yes | sdkmanager --licenses` once and skip license request further
* why run several commands when `sdkmanager --install <pkg> <pkg> <pkg> ...` form can be used

Reply:
Applied.